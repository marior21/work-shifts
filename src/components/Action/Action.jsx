import { Pressable, Text, styles } from '@mirai/ui';
import PropTypes from 'prop-types';
import React from 'react';

import * as style from './Action.module.css';

export const Action = ({ button, children, onPress, ...others }) => (
  <Pressable
    data-testid={others['data-testid']}
    type="button"
    onPress={onPress}
    className={styles(button && style.button, others.className)}
  >
    <Text {...others} data-testid="">
      {children}
    </Text>
  </Pressable>
);

Action.propTypes = {
  button: PropTypes.bool,
  children: PropTypes.node.isRequired,
  onPress: PropTypes.func,
};
