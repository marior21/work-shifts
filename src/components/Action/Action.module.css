.button {
  width: 100%;
  padding: 0 var(--mirai-ui-space-XS);
}
