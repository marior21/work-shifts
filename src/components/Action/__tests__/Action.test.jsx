import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import { Action } from '../Action';

const DEFAULTS = {
  children: 'Lorem Ipsum...',
};

describe('components/ <Action>', () => {
  test('renders', () => {
    const { asFragment } = render(<Action {...DEFAULTS} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('prop:button', () => {
    const { asFragment } = render(<Action {...DEFAULTS} button />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('prop:onPress', () => {
    const handlePress = jest.fn();
    const testId = 'onChange';
    const { getByTestId } = render(<Action {...DEFAULTS} onPress={handlePress} data-testid={testId} />);

    fireEvent.click(getByTestId(testId));
    expect(handlePress).toHaveBeenCalledTimes(1);
  });

  test('extending properties from <Text>', () => {
    const { asFragment } = render(<Action {...DEFAULTS} small />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Testing
  test('testID', () => {
    const { asFragment, getByTestId } = render(<Action {...DEFAULTS} data-testid="mirai" />);
    expect(getByTestId('mirai')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  // Styling
  test('inherit:className', () => {
    const { asFragment } = render(<Action {...DEFAULTS} className="mirai" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
