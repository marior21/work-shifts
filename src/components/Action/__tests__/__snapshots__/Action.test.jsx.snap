// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ <Action> extending properties from <Text> 1`] = `
<DocumentFragment>
  <button
    class="pressable"
    type="button"
  >
    <span
      class="text small"
      data-testid=""
    >
      Lorem Ipsum...
    </span>
  </button>
</DocumentFragment>
`;

exports[`components/ <Action> inherit:className 1`] = `
<DocumentFragment>
  <button
    class="pressable mirai"
    type="button"
  >
    <span
      class="text paragraph mirai"
      data-testid=""
    >
      Lorem Ipsum...
    </span>
  </button>
</DocumentFragment>
`;

exports[`components/ <Action> prop:button 1`] = `
<DocumentFragment>
  <button
    class="pressable"
    type="button"
  >
    <span
      class="text paragraph"
      data-testid=""
    >
      Lorem Ipsum...
    </span>
  </button>
</DocumentFragment>
`;

exports[`components/ <Action> renders 1`] = `
<DocumentFragment>
  <button
    class="pressable"
    type="button"
  >
    <span
      class="text paragraph"
      data-testid=""
    >
      Lorem Ipsum...
    </span>
  </button>
</DocumentFragment>
`;

exports[`components/ <Action> testID 1`] = `
<DocumentFragment>
  <button
    class="pressable"
    data-testid="mirai"
    type="button"
  >
    <span
      class="text paragraph"
      data-testid=""
    >
      Lorem Ipsum...
    </span>
  </button>
</DocumentFragment>
`;
